agent/api/task.go

```diff
@@ -110,6 +111,13 @@ type Task struct {
 	// used to look up the credentials for task in the credentials manager
 	credentialsID     string
 	credentialsIDLock sync.RWMutex
+
+	// cgroupSpecLock to reliably update the cgroup spec
+	cgroupSpecLock sync.RWMutex
+
+	// CgroupSpec attribute to capture the cgroup spec
+	// TODO: Inspect upon model changes
+	CgroupSpec *cgroup.Spec `json:omitempty`
```

Comment:
Can you order this like everything else in the file?  That means moving `cgroupSpecLock` directly below `CgroupSpec`.

Also, we discussed putting the `CgroupSpec` inside a platform-specific wrapper struct such that it wouldn't compile in on Windows.  Is there a reason you went away from that?